src/main/java/de/tum/in/www1/artemis/service/scheduled/AutomaticSubmissionService.java

```diff
@@ -119,10 +119,13 @@ private Submission updateParticipation(Submission submission) {
             if (submission instanceof ModelingSubmission) {
                 ModelingExercise modelingExercise = (ModelingExercise) exercise;
                 ModelingSubmission modelingSubmission = (ModelingSubmission) submission;
-                // notify compass about new submission
-                modelingSubmissionService.notifyCompass(modelingSubmission, modelingExercise);
-                // check if compass could assess automatically
-                modelingSubmissionService.checkAutomaticResult(modelingSubmission, modelingExercise);
+                try {
+                    // notify compass about new submission
+                    modelingSubmissionService.notifyCompass(modelingSubmission, modelingExercise);
```

Comment:
I think this should be its own method.
At least I see no connection of notifying compass about the submission in a method called `updateParticipation`.

If we keep it this way, the side effect should be mentioned in the JavaDoc.

Reply:
Well, it says `In the case of a modeling exercise Compass is additionally notified about the new submission.` But you are completely right, notifying Compass has nothing to do with `updateParticipation`. I created another private method `notifyCompassAboutNewModelingSubmission` that is called from the `run` method above, right **after** the submission is saved to the database.